### PR TITLE
[FEATURE] ModifyRedirectUrlValidationResultEvent

### PR DESCRIPTION
**Pull request by linawolf**
Releases: main
Resolves: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/911

### CODE REVIEW

**Comment by linawolf** [comment]
@brotkrueml I think as the event only exists in TYPO3 13 we can start with Atrtibute right away?

File: Documentation/ApiOverview/Events/Events/FrontendLogin/ModifyRedirectUrlValidationResultEvent.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    results for the redirect URL, allowing redirects to URLs not matching the
+    existing validation constraints.
+
+The PSR-14 event
+:php:`\TYPO3\CMS\FrontendLogin\Event\ModifyRedirectUrlValidationResultEvent`
+provides developers with the possibility and flexibility to implement custom
+validation for the redirect URL in the frontend login.
+
+This may be useful, if TYPO3 frontend login
+acts as an SSO system or if users should be redirected to an external URL after
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Events/Events/FrontendLogin/ModifyRedirectUrlValidationResultEvent.rst:20]:
```suggestion
acts as an :abbr:`SSO (Single-Sign On)` system, or if users should be redirected to an external URL after
```

File: Documentation/ApiOverview/Events/Events/FrontendLogin/_ModifyRedirectUrlValidationResultEvent/_ValidateRedirectUrl.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Vendor\MyExtension\EventListener;
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Events/Events/FrontendLogin/_ModifyRedirectUrlValidationResultEvent/_ValidateRedirectUrl.php:5]:
```suggestion
namespace MyVendor\MyExtension\EventListener;
```

File: Documentation/ApiOverview/Events/Events/FrontendLogin/_ModifyRedirectUrlValidationResultEvent/_ValidateRedirectUrl.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+use TYPO3\CMS\Core\Attribute\AsEventListener;
+use TYPO3\CMS\FrontendLogin\Event\ModifyRedirectUrlValidationResultEvent;
+
+final class ValidateRedirectUrl
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Events/Events/FrontendLogin/_ModifyRedirectUrlValidationResultEvent/_ValidateRedirectUrl.php:10]:
```suggestion
final readonly class ValidateRedirectUrl
```

File: Documentation/ApiOverview/Events/Events/FrontendLogin/_ModifyRedirectUrlValidationResultEvent/_ValidateRedirectUrl.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public const TRUSTED_HOST_FOR_REDIRECT = 'example.org';
+
+    #[AsEventListener('validate-custom-redirect-url')]
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Events/Events/FrontendLogin/_ModifyRedirectUrlValidationResultEvent/_ValidateRedirectUrl.php:14]:
```suggestion
    #[AsEventListener(
        identifier: 'validate-custom-redirect-url'
    )]
```

File: Documentation/ApiOverview/Events/Events/FrontendLogin/_ModifyRedirectUrlValidationResultEvent/_ValidateRedirectUrl.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+declare(strict_types=1);
+
+namespace Vendor\MyExtension\EventListener;
+
+use TYPO3\CMS\Core\Attribute\AsEventListener;
+use TYPO3\CMS\FrontendLogin\Event\ModifyRedirectUrlValidationResultEvent;
+
+final class ValidateRedirectUrl
+{
+    public const TRUSTED_HOST_FOR_REDIRECT = 'example.org';
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Events/Events/FrontendLogin/_ModifyRedirectUrlValidationResultEvent/_ValidateRedirectUrl.php:12]:
```suggestion
    private const TRUSTED_HOST_FOR_REDIRECT = 'example.org';
```

File: Documentation/ApiOverview/Events/Events/FrontendLogin/_ModifyRedirectUrlValidationResultEvent/_ValidateRedirectUrl.php (hunk 3 above)
Comment (linawolf) [reply Documentation/ApiOverview/Events/Events/FrontendLogin/_ModifyRedirectUrlValidationResultEvent/_ValidateRedirectUrl.php]: None of the other Event Listeners I looked at is marked as readonly. Should we mass-change them?

Comment (brotkrueml) [reply Documentation/ApiOverview/Events/Events/FrontendLogin/_ModifyRedirectUrlValidationResultEvent/_ValidateRedirectUrl.php]: Hmm, thought I've already done that. Will do.

**Review by brotkrueml — APPROVED** [review]

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**